wordpress-mobile/WordPress-Android
# My Site Dashboard: Tabs - Refactor MySiteFragment to support tabs

Fixes #16048
Fixes #16003 

This PR adds "tab" functionality to the My Site Fragment

Menu | Dashboard
-- | --
<img width="250" height="500" alt="Menu" src="https://user-images.githubusercontent.com/506707/158202892-1a27713a-6b5d-4287-a3e1-80d9614c64e5.png"> | <img width="250" height="500" alt="Dashboard" src="https://user-images.githubusercontent.com/506707/158202872-871745f8-b63a-40a5-bbd1-d3aa98b31d20.png">

1. Adds MySiteTabFragment
2. Reuses MySiteViewModel for each tab
3. Maintains PTR in each tab
4. Builds distinct views for each tab 
5. Maintains a single view containing all cards/site items (for when tab feature flag is disabled)

Known issues/not included in this PR
1. Dynamic card left/right swiping may swipe entire tab 
2. Moving site header above tabs
3. Tab labels 
4. PTR as a single unit across both tabs
5. While on Dashboard tab, notification dot not shown on menu tab. See comment [below](https://github.com/wordpress-mobile/WordPress-Android/pull/16099#pullrequestreview-910034172).

@tiagomar - Would love your eyes on this one - most important items are tests A, B, and C, with Quick Start as the highest priority. Thanks.

**To test:**

The first set of tests focus on regression testing functionality while tabs are disabled
Prerequisite
1. Launch the app.
6. Go to `My Site` -> `Me` -> `App Settings` -> `Debug settings`.
7. Turn off the `MySiteDashboardTabsFeatureConfig` flag under the `Features in development` section.
8. Restart the app
9. Logout/login and select "Show me around" when the quick start prompt is shown.

**Test A: Regression test Quick Start in single view (tabs disabled)**
1. Run tests in [wpandroid-quick-start-testing-instructions.txt](https://github.com/wordpress-mobile/WordPress-Android/files/8246096/wpandroid-quick-start-testing-instructions.txt)

**Test B: Regression test general navigation (tabs disabled)**
1. Navigate around the My Site View and validate that navigation works as expected (FAB, site items, Quick links)

**Test C: Regression test - PTR (tabs disabled)**
1. Swipe down on the My Site view to request a refresh of the view
----------------------------

The remainder of the tests focus on testing functionality while tabs are enabled
Prerequisite
1. Launch the app.
3. Go to `My Site` -> `Me` -> `App Settings` -> `Debug settings`.
4. Turn on the `MySiteDashboardTabsFeatureConfig` flag under the `Features in development` section.
6. Restart the app

**Test D: Quick Start 
1. Logout/login and select "Show me around" when the quick start prompt is shown.
2. Note the "Quick Start" shows in the "Menu" tab only 
3. Run tests in [wpandroid-quick-start-testing-instructions.txt](https://github.com/wordpress-mobile/WordPress-Android/files/8246096/wpandroid-quick-start-testing-instructions.txt)

**Test E: General navigation**
1. Navigate around the My Site View and validate that navigation works as expected (FAB, site items, Quick links)

**Test F: Content validation**
1. Validate the following tab content:
Menu tab includes: Site Info header, quick links, quick start (if selected), domain registration card (if available), and site menu items.
Dashboard tab includes: Site info header, quick links, domain registration card (if available), and dashboard cards (stats, posts).


## Regression Notes
1. Potential unintended areas of impact
Quick start actions/tasks don't work as expected

4. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and automated tested

10. What automated tests I added (or what prevented me from doing so)
Unit tests were updated to reflect refactoring efforts


PR submission checklist:
